include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -135,6 +140,294 @@ class alignment_file_format_sam
+                }
+            }
+            else
+            {
+                ++begin(stream_view); // skip '*'
+            }
+        }
+        else
+        {
+            detail::consume(stream_view | next);
+        }
+        offset = offset_tmp;
+        ++begin(stream_view);
+
+        // Field 7-9: (RNEXT PNEXT TLEN)=MATE
+        // -------------------------------------------------------------------------------------------------------------
+        if constexpr (!detail::decays_to_ignore_v<mate_type>)
+        {
+            read_field(stream_view | next, get<0>(mate)); // RNEXT
+            ++begin(stream_view);
+
+            read_field(stream_view | next, get<1>(mate)); // PNEXT
+            ++begin(stream_view);
+
+            read_field(stream_view | next, get<2>(mate)); // TLEN
+            ++begin(stream_view);
+        }
+        else
+        {
+            for (size_t i = 0; i < 3; ++i)
+            {
+                detail::consume(stream_view | next);
+                ++begin(stream_view);
+            }
+        }
+
+        // Field 10: Sequence
+        // -------------------------------------------------------------------------------------------------------------
+        if (!is_char<'*'>(*std::ranges::begin(stream_view))) // sequence information is given
+        {
+            auto constexpr is_legal_alph = is_in_alphabet<seq_legal_alph_type>;
+            auto seq_stream = stream_view | next
+                                          | view::transform([is_legal_alph] (char const c) // enforce legal alphabet
+                                            {
+                                                if (!is_legal_alph(c))
+                                                    throw parse_error{std::string{"Encountered an unexpected letter: "} +
+                                                                      is_legal_alph.msg.string() +
+                                                                      " evaluated to false on " +
+                                                                      detail::make_printable(c)};
+                                                return c;
+                                            });
+
+            if constexpr (!detail::decays_to_ignore_v<align_type> && detail::decays_to_ignore_v<seq_type>)
+            {
+                if (!cigar.empty()) // only parse alignment if cigar information was given
+                {
+                    static_assert(sequence_container_concept<std::remove_reference_t<decltype(get<1>(align))>>,
+                                  "If you want to read ALIGNMENT but not SEQ, the alignment"
+                                  " object must store a sequence container at the second position.");
+
+                    // detail::consume(seq_stream | view::take_exactly_or_throw(offset_tmp));        // skip soft clipping at begin
+                    for (; offset_tmp > 0; --offset_tmp) // offset_tmp is not needed anymore
```

Comment:
does not work.. something along ` cannot bind non-const lvalue reference of [very long ranges type of seq_stream iterator &] .. to an rvalue of [very long type of seq_stream iterator] `.....

Reply:
Yes, because 
`std::ranges::advance(std::ranges::begin(seq_stram),  offset_tmp)` won't work for the given reason.
I think even if it does not safe loc:
```cpp
auto tmp_it = std::ranges::begin(seq_stream);
std::ranges::advance(tmp_it,  offset_tmp);
```
is still better practice.

Reply:
fine